app/models/work.rb

```diff
@@ -1146,7 +1151,7 @@ def mark_as_ham!
   def notify_of_hiding
     return unless hidden_by_admin? && saved_change_to_hidden_by_admin?
     users.each do |user|
-      if spam?
+      if spam? && hidden_for_spam
```

Comment:
```suggestion
      if spam? && saved_change_to_spam?
```
This should achieve the same thing as the extra attribute, but it's not as fragile as an attribute that can be forgotten about.